From e90f2ba11bda106b063b933317c6d2079e591266 Mon Sep 17 00:00:00 2001
From: Duy Dang <duy.dang.yb@renesas.com>
Date: Wed, 22 Jul 2020 13:32:06 +0700
Subject: [PATCH 283/476] ARM: dts: r8a7745: Add VCP/VPC nodes

Add vcp/vpc nodes for Renesas uvcs-kernel-module probe.

Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
---
 arch/arm/boot/dts/r8a7745.dtsi | 48 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7745.dtsi b/arch/arm/boot/dts/r8a7745.dtsi
index c9a50db..472e4b4 100644
--- a/arch/arm/boot/dts/r8a7745.dtsi
+++ b/arch/arm/boot/dts/r8a7745.dtsi
@@ -1009,6 +1009,54 @@
 			status = "disabled";
 		};
 
+		vcp0: vcp0@fe900000 {
+			compatible = "renesas,vcp0";
+			reg = <0 0xfe900000 0 0x200>, <0 0xfe900200 0 0x200>;
+			interrupts = <0 258 IRQ_TYPE_LEVEL_HIGH>,
+				<0 259 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7745_CLK_VCP0>;
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc0: vcp0@fe908000 {
+			compatible = "renesas,vpc0";
+			reg = <0 0xfe908000 0 0x90>;
+			clocks = <&mstp1_clks R8A7745_CLK_VPC0>;
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc1: vcp0@fe918000 {
+			compatible = "renesas,vpc1";
+			reg = <0 0xfe918000 0 0x90>;
+			/* clocks = <&mstp1_clks R8A7745_CLK_VPC1>; */
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc0xy: vcp0@fe960380 {
+			compatible = "renesas,vpc0xy";
+			reg = <0 0xfe960380 0 0x4>;
+			/* clocks = <&mstp1_clks R8A7745_CLK_VPC0>; */
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc1xy: vcp0@fe960384 {
+			compatible = "renesas,vpc1xy";
+			reg = <0 0xfe960384 0 0x4>;
+			/* #clocks = <&mstp1_clks R8A7745_CLK_VPC1>; */
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		prr: vcp0@ff000044 {
+			compatible = "renesas,prr";
+			reg = <0 0xff000044 0 0x4>;
+			renesas,#ch = <0>;
+		};
+
 		rcar_sound: sound@ec500000 {
 			/*
 			 * #sound-dai-cells is required
-- 
2.7.4

